### ISSUE TITLE
Question regarding beta 6

### ISSUE THREAD

**Issue by CooperCGN** [naofireblade/homebridge-neato#68]
I know it’s beta but I was running beta 4 already quite successfully. Just updated to beta 6 and now all I get in the home app is two of the house icons stating being not supported. All they contain is the battery status, no switches to start cleaning.

Do you have a rough idea (not trying to push you) how further development will go? I might consider going to the latest stable depending on that.

**Comment by naofireblade** [comment]
I will try to release a stable beta today evening. 

The room cleaning feature may be included as well but is not tested.

**Comment by CooperCGN** [comment]
That sounds awesome, I personally don’t need the room cleaning feature right now. Just want to start whole map cleaning when leaving the house.

**Comment by DJay-X** [comment]
> I personally d̶o̶n̶t̶ need the room cleaning feature 
😅

@naofireblade I will of course test the feature when available in the new beta. 
https://github.com/naofireblade/homebridge-neato/issues/44#issuecomment-840343350

**Comment by naofireblade** [comment]
The weather was to good today. Next attempt for a new beta tomorrow :) .

**Comment by naofireblade** [comment]
Hi, I published beta 7. If it still does not work you need to remove your robot(s) from cache in Homebridge UI:

![image](https://user-images.githubusercontent.com/12081369/119535036-e11de700-bd87-11eb-951c-edf144b993c4.png)

Room cleaning is not ready for testing yet but the rooms should be listed in your log.

**Comment by naofireblade** [comment]
> > I personally d̶o̶n̶t̶ need the room cleaning feature
> > 😅
> 
> @naofireblade I will of course test the feature when available in the new beta.
> [#44 (comment)](https://github.com/naofireblade/homebridge-neato/issues/44#issuecomment-840343350)

That would be awesome! I'm planning to make an option for queuing the rooms instead of interrupting the currently cleaned one.

**Comment by CooperCGN** [comment]
So everything is working for me except the zone cleaning where I have no switches showing up. Maybe it’s because I have a D5 and a D7? D5 doesn’t support zones.

**Comment by naofireblade** [comment]
Zones (rooms) are not available in the beta yet. Could you check if your log lists your rooms after removing the cached accessory? That would help me to know if I'm on the right path.

**Comment by CooperCGN** [comment]
Yes they’re showing up with the remark that not available in beta 😂 stupid me

**Comment by naofireblade** [comment]
> Yes they’re showing up with the remark that not available in beta 😂 stupid me

Great. Thanks!

**Comment by DJay-X** [comment]
Thank a lot. Already installed beta 7.
I noticed that in Eve App something is wrong with the switches "Sauge Bereich" and "Saugen". They both seem linked together somehow. I think I can remember we had something similar in the past. I can't sort/arrange them in Eve App in a different order. Also when changing the Icons on one switch it also changes on other switches as well. 
Everything else with my D7 is looking good for now I think.

```
[26.5.2021, 12:44:09] [NeatoVacuumRobot] [Neato D7] has [object Object] saved map.
[26.5.2021, 12:44:09] [NeatoVacuumRobot] [Neato D7] Found floorplan {"id":"2020-04-16T12:09:25Z","name":"Wohnung","url":"https://neatorobotics.s3.amazonaws.com/persistent-maps/xxxxx/2020-04-16T12:09:25Z/xxxxxx/user-map.png?X-Amz-Algorithm=xxxxxX-Amz-Credential=xxxxx%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210526T104409Z&X-Amz-Expires=300&X-Amz-SignedHeaders=host&X-Amz-Signature=xxxxx","raw_floor_map_url":"https://neatorobotics.s3.amazonaws.com/persistent-maps/xxxxx-xxxxx/2020-04-16T12:09:25Z/xxxxx/floor-map.png?X-Amz-Algorithm=xxxx&X-Amz-Credential=xxxx%xxxx%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=xxxx&X-Amz-Expires=300&X-Amz-SignedHeaders=host&X-Amz-Signature=xxxxxx","url_valid_for_seconds":300}
[26.5.2021, 12:44:12] [NeatoVacuumRobot] [Neato D7] Found room Schlafzimmer. Not adding in current beta.
[26.5.2021, 12:44:12] [NeatoVacuumRobot] [Neato D7] Found room Wohnzimmer. Not adding in current beta.
[26.5.2021, 12:44:12] [NeatoVacuumRobot] [Neato D7] Found room Esszimmer. Not adding in current beta.
[26.5.2021, 12:44:12] [NeatoVacuumRobot] [Neato D7] Found room Küche. Not adding in current beta.
[26.5.2021, 12:44:12] [NeatoVacuumRobot] [Neato D7] Found room Flur. Not adding in current beta.
[26.5.2021, 12:44:12] [NeatoVacuumRobot] [Neato D7] Found room Badezimmer. Not adding in current beta.
[26.5.2021, 12:44:12] [NeatoVacuumRobot] [Neato D7] Successfully created as new robot
```

**Comment by CooperCGN** [comment]
Good morning, before you read the rest, is the background polling already implemented while a robot is cleaning?

I did some further testing with automations. If I understand the basics right the plugin is supposed to switch automatically between updating the status every 30 minutes while a robot is inactive and updating every 60 seconds while cleaning until he returns to the base.

While my automation which uses starting the upstairs robot as trigger (either by Siri or by automation) works perfectly the other one using the charging base being occupied again after cleaning only triggered like 15 to 20 minutes after it actually stopped.

**Comment by DJay-X** [comment]
@naofireblade I hope everything is all right with you.
Any news with the ongoing development of the beta? Looking forward to room cleaning feature again.
The basics with v1.0.0-beta.7 are working good for me since a month now.
Thanks again.

**Comment by naofireblade** [comment]
Sure, just a lot of work to do. Thanks for the little reminder here, I will look to find some free time for the plugin

**Comment by CooperCGN** [comment]
Is this still a thing? 😄

**Comment by naofireblade** [comment]
I‘m still alive 😊 and have a bad conscience because of the halting development. Hope I will have some time soon.

**Comment by CooperCGN** [comment]
Just to let you know: my D5 works as expected but every time I start my D7 from HomeKit after a while it stops and says it is on the wrong floor. Even more weird, I then have to factory reset it completely because it won’t recognize the floor plan anymore. I ran it over months directly from the app and it works flawlessly. Maybe your plugin sends some incorrect data?
